### PR TITLE
feat(completion): added `avante-nvim` plugin

### LINKED ISSUE #1168: [NEW] Avante.nvim
### Is your feature related to a problem?

No

### Describe the new feature

https://github.com/yetone/avante.nvim

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by vonhyou**
## 📑 Description

Add a new plugin for a cursor-like experience, supports Claude, OpenAI, Copilot etc.

Closes #1168

### CODE REVIEW

File: lua/astrocommunity/completion/avante-nvim/init.lua
Comment (luxus) [inline lua/astrocommunity/completion/avante-nvim/init.lua:14]: please use mini.icons

File: lua/astrocommunity/completion/avante-nvim/init.lua
Comment (luxus) [inline lua/astrocommunity/completion/avante-nvim/init.lua:4]: can be removed if command and keybindings are added

File: lua/astrocommunity/completion/avante-nvim/init.lua
Comment (vonhyou) [reply lua/astrocommunity/completion/avante-nvim/init.lua]: sry, fixed in the latest commit.

File: lua/astrocommunity/completion/avante-nvim/init.lua
Comment (vonhyou) [reply lua/astrocommunity/completion/avante-nvim/init.lua]: removed in the latest commit

**Comment by luxus** [comment]
https://github.com/yetone/avante.nvim/wiki#keymaps-and-api-i-guess
keymaps should be done with astrocore i guess
and should load on avante commands and on entering visual mode

**Comment by kaka-ruto** [comment]
Thanks @vonhyou !

File: lua/astrocommunity/completion/avante-nvim/README.md
Comment (Uzaaft) [inline lua/astrocommunity/completion/avante-nvim/README.md:21]:
```suggestion
Everytime you open a file, the plugin will ask you to enter an Anthropic API key for Claude 3.5 Sonnet. To avoid repeating this, set an environment variable to store your API key locally:

For Claude:

```sh
export ANTHROPIC_API_KEY=your-api-key
```

For OpenAI:

```sh
export OPENAI_API_KEY=your-api-key
```

For Azure OpenAI:

```sh
export AZURE_OPENAI_API_KEY=your-api-key
```
```

I believe we can just leave this out and point to their docs due to the alpha nature of the plugin.  Thoughts? @luxus

File: lua/astrocommunity/completion/avante-nvim/init.lua
Comment (Uzaaft) [inline lua/astrocommunity/completion/avante-nvim/init.lua:3]:
```suggestion
  "yetone/avante.nvim",
  event = "VeryLazy",
  lazy = false,
  build = ":AvanteBuild",
```

Added in event and lazy = false from the docs.

File: lua/astrocommunity/completion/avante-nvim/init.lua
Comment (Uzaaft) [inline lua/astrocommunity/completion/avante-nvim/init.lua:20]:
```suggestion
```
This can be removed if `lazy = false` is added

File: lua/astrocommunity/completion/avante-nvim/init.lua
Comment (vonhyou) [reply lua/astrocommunity/completion/avante-nvim/init.lua:3]: sry for the delay. thank you for your review. i'm not sure which way the project prefer since i was asked to remove these two lines and add commands before. so i'll leave it to you maintainers first. have a nice day

Comment (Uzaaft) [reply lua/astrocommunity/completion/avante-nvim/init.lua:3]: Does the loading work in the newest spec? @vonhyou . From my own experience, it might not when you're using plugins written in multiple languages.

Comment (vonhyou) [reply lua/astrocommunity/completion/avante-nvim/init.lua:3]: Hi, when I tested in my environment, the plugin did load after entering the cmds according to lazy profile. But I never tried windows or other platforms. I think it's reasonable to go with your way

File: lua/astrocommunity/completion/avante-nvim/init.lua
Comment (luxus) [reply lua/astrocommunity/completion/avante-nvim/init.lua]: whats the reason behind `lazy = false` ? it works fine without it.. and it imho it should follow the user config  and not force loading on startup (it will load a lot deps as well)

File: lua/astrocommunity/completion/avante-nvim/init.lua
Comment (luxus) [inline lua/astrocommunity/completion/avante-nvim/init.lua:58]: deps will always get loaded

File: lua/astrocommunity/completion/avante-nvim/init.lua
Comment (Uzaaft) [reply lua/astrocommunity/completion/avante-nvim/init.lua]: > whats the reason behind `lazy = false` ? it works fine without it.. and it imho it should follow the user config and not force loading on startup (it will load a lot deps as well)

The docs dosent have lazy = true, they recommend lazy = false

File: lua/astrocommunity/completion/avante-nvim/init.lua
Comment (vonhyou) [reply lua/astrocommunity/completion/avante-nvim/init.lua]: fixed

File: lua/astrocommunity/completion/avante-nvim/init.lua
Comment (vonhyou) [reply lua/astrocommunity/completion/avante-nvim/init.lua:20]: hey @luxus and @Uzaaft, i've been testing the plugin with the current configuration (leaving out `lazy = false` and define `cmd`s). loading on commands can reduce about 20ms start time at most on my end and it's working. if the speed matters i believe it's ready to go as it is.

Comment (Uzaaft) [reply lua/astrocommunity/completion/avante-nvim/init.lua:20]: I'm going to defer that decision to luxux. but this can break in the future.

Comment (vonhyou) [reply lua/astrocommunity/completion/avante-nvim/init.lua:20]: @Uzaaft thank you for your help these days.

File: lua/astrocommunity/completion/avante-nvim/init.lua
Comment (mehalter) [inline lua/astrocommunity/completion/avante-nvim/init.lua:68]:
```suggestion
```

this should be removed as `nvim-web-devicons` is default in AstroNvim and will be up to the user to change it.

File: lua/astrocommunity/completion/avante-nvim/init.lua
Comment (mehalter) [inline lua/astrocommunity/completion/avante-nvim/init.lua:69]:
```suggestion
```

This should be left up to the user

File: lua/astrocommunity/completion/avante-nvim/init.lua
Comment (mehalter) [inline lua/astrocommunity/completion/avante-nvim/init.lua:94]: These are optional and should probably be left to the user

**Review by mehalter — CHANGES_REQUESTED** [review]

File: lua/astrocommunity/completion/avante-nvim/init.lua
Comment (vonhyou) [reply lua/astrocommunity/completion/avante-nvim/init.lua]: hi @mehalter, thank you for your comment. i was asked to switch to mini.icons here before <https://github.com/AstroNvim/astrocommunity/pull/1184#discussion_r1740203564>. but your comment makes sense. committed.

File: lua/astrocommunity/completion/avante-nvim/init.lua
Comment (mehalter) [inline lua/astrocommunity/completion/avante-nvim/init.lua:68]:
```suggestion
  opts = {},
  specs = { -- configure optional plugins
    { -- if copilot.lua is available, default to copilot provider
      "zbirenbaum/copilot.lua",
      optional = true,
      specs = {
        {
          "yetone/avante.nvim",
          opts = {
            provider = "copilot",
          },
        },
      },
    },
    {
      -- make sure `Avante` is added as a filetype
      "MeanderingProgrammer/render-markdown.nvim",
      optional = true,
      opts = function(_, opts)
        if not opts.file_types then opts.filetypes = { "markdown" } end
        table.insert(opts.file_types, "Avante")
      end,
    },
  },
```

We can actually add in some optional specs that will help configure optional plugins if they are available!

**Review by mehalter — CHANGES_REQUESTED** [review]
This looks great! One last thing is we can add some extra stuff to help configure optional plugins :)

File: lua/astrocommunity/completion/avante-nvim/init.lua
Comment (mehalter) [reply lua/astrocommunity/completion/avante-nvim/init.lua:68]: Also added some more AstroCommunity entries to make it easy for users to opt into these options with good lazy loading and defaults as well:

https://github.com/AstroNvim/astrocommunity/pull/1193
https://github.com/AstroNvim/astrocommunity/pull/1194

**Review by mehalter — APPROVED** [review]

**Pull request merged**
